zio-http/shared/src/main/scala/zio/http/Handler.scala

```diff
@@ -746,10 +746,49 @@ object Handler extends HandlerPlatformSpecific with HandlerVersionSpecific {
     fromResponse(Response.error(status))
 
   /**
-   * Creates a handler with an error and the specified error message.
+   * Creates a handler with an error and the specified error message. The error
+   * message will be returned as HTML, JSON or plain text depending on the
+   * `Accept` header of the request, defaulting to
+   * [[ErrorResponseConfig.errorFormat]]. If
+   * [[ErrorResponseConfig.withErrorBody]] is `false`, the error message will
+   * not be included in the response. If the error message should always be
+   * included in the response, use `Handler.fromResponse(Response.error(status,
+   * message))`
    */
   def error(status: => Status.Error, message: => String): Handler[Any, Nothing, Any, Response] =
-    fromResponse(Response.error(status, message))
+    (fromResponse(Response.status(status)) @@ Middleware.interceptHandlerStateful(
+      handler((req: Request) => (req.header(Header.Accept), (req, ()))),
+    ) {
+      handler { (accept: Option[Header.Accept], res: Response) =>
+        ErrorResponseConfig.configRef.get.map { cfg =>
+          if (cfg.withErrorBody) {
+            val mediaType: MediaType = accept
+              .flatMap(_.mimeTypes.sortBy(mt => -mt.qFactor.getOrElse(1.0)).map(_.mediaType).collectFirst {
```

Comment:
Should we add an implicit `Ordering` in the companion object of `Accept`? 🤔

Reply:
We could also ensure it is always sorted.

Reply:
Best to ensure it's always sorted, we don't want to be doing this on each request.

Reply:
It has to be done on each request, since this the the requests accept header. But since the amount of accepted media types will be small, I thought a sorted list might be easier to understand. It is just sorted by q factor when ever you deal with the typed interface.

Reply:
Maybe you could check and avoid sorting it if it's already sorted, which would avoid the overhead entirely for the happy path.